File: problem4.py
#Time Complexity - 
#push - O(1)
#pop - O(n)
#peek - O(1)
#empty - O(1)

#Space Complexity - O(n)

#Did this code successfully run on Leetcode : Yes
#Any problem you faced while coding this : No

class MyQueue:

    def __init__(self):
        self.queue=[]

    def push(self, x: int) -> None:
        self.queue.append(x)

    def pop(self) -> int:
        if len(self.queue)>0:
            return self.queue.pop(0)

    def peek(self) -> int:
        return self.queue[0]

    def empty(self) -> bool:
        if self.queue==[]:
            return True
        return False


        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)

File: Problem4.java
// Time Complexity :
// Space Complexity :
// Did this code successfully run on Leetcode :
// Any problem you faced while coding this :


// Your code here along with comments explaining your approach
//Design a hash map

class MyHashMap {
    
      LinkedList<Entry>[] map;
    int SIZE=769; //prime number to reduce chances of collision

    public MyHashMap() {
        map=new LinkedList[SIZE];
        
    }
    
    public void put(int key, int value) {
        
         int bucket=key%SIZE;
        if(map[bucket]==null){
            map[bucket]=new LinkedList<Entry>();
            map[bucket].add(new Entry(key,value));
        }else{
            
            for (Entry entry: map[bucket]){
                
                if(entry.key==key)return;
            }
        }
        map[bucket].add(new Entry(key,value));
        
    }
    
    public int get(int key) {
        
        int bucket=key%SIZE;
        LinkedList<Entry> entries=map[bucket];
        if (entries==null) return -1;
        for (Entry entry:entries){
            if(entry.key==key) return key;
        }
        return -1;
         }
        
    
    public void remove(int key) {
        int bucket = key % SIZE;
			Entry toRemove = null;
			if(map[bucket] == null) return;
			else {
				for(Entry entry : map[bucket]){
					if(entry.key == key) {
						toRemove = entry;
					}
				}
				if(toRemove == null) return;
				map[bucket].remove(toRemove);
			}
    }}
class Entry{
        int key;
        int val;
        
        Entry(int key, int val){
            this.key=key;
            this.val=val;
        
    }
}
